src/youtube.js

```diff
@@ -0,0 +1,1092 @@
+   * 0 = HAVE_NOTHING - no information whether or not the audio/video is ready.
+   * 1 = HAVE_METADATA - metadata for the audio/video is ready.
+   * 2 = HAVE_CURRENT_DATA - data for the current playback position is available, but not enough data to play next frame/millisecond.
+   * 3 = HAVE_FUTURE_DATA - data for the current and at least the next frame is available.
+   * 4 = HAVE_ENOUGH_DATA - enough data available to start playing.
+   */
+  get readyState(): number {
+    if (!this._api) {
+      return 0;
+    }
+    const playerState = window.YT.PlayerState;
+    switch (this._api.getPlayerState()) {
+      case playerState.UNSTARTED:
+        return 0;
+      case playerState.BUFFERING:
+        return 2;
+      default:
+        return 4;
+    }
+  }
+
+  /**
+   * @returns {Number} - The height of the video player, in pixels.
+   * @public
+   */
+  get videoHeight(): number {
+    return -1;
+  }
+
+  /**
+   * @returns {Number} - The width of the video player, in pixels.
+   * @public
+   */
+  get videoWidth(): number {
+    return -1;
+  }
+
+  /**
+   * @param {boolean} playsinline - Whether to set on the video tag the playsinline attribute.
+   */
+  set playsinline(playsinline: boolean): void {}
+
+  /**
+   * @returns {boolean} - Whether the video tag has an attribute of playsinline.
+   */
+  get playsinline(): boolean {
+    return this._config.playsinline;
+  }
+
+  /**
+   * Set crossOrigin attribute.
+   * @param {?string} crossOrigin - 'anonymous' or 'use-credentials'
+   */
+  set crossOrigin(crossOrigin: ?string): void {}
+
+  /**
+   * Get crossOrigin attribute.
+   * @returns {?string} - 'anonymous' or 'use-credentials'
+   */
+  get crossOrigin(): ?string {
+    return null;
+  }
+
+  /**
+   * get the playback rates
+   * @return {number[]} - playback rates
+   */
+  get playbackRates(): Array<number> {
+    return this._api ? this._api.getAvailablePlaybackRates() : Youtube.PLAYBACK_RATES;
+  }
+
+  /**
+   * Initializes the engine.
+   * @param {PKMediaSourceObject} source - The selected source object.
+   * @param {Object} config - The player configuration.
+   * @private
+   * @returns {void}
+   */
+  _init(source: PKMediaSourceObject, config: Object): void {
+    this._source = source;
+    this._config = config;
+    this._sdkLoaded = new Promise((resolve, reject) => {
+      this._apiReady = () => {
+        if (this._config.playback.muted) {
+          this.muted = true;
+        }
+        resolve();
+      };
+      this._apiError = () => {
+        //todo, throw error
+        reject();
+      };
+    });
+    this._loadYouTubeIframeAPI().then(() => {
+      this._loadYouTubePlayer();
+    });
+
+    // this.attach();
+  }
+
+  /**
+   * Loads the YouTube player.
+   * @private
+   * @returns {void}
+   */
+  _loadYouTubePlayer(): void {
+    const loadYouTubePlayer = () => {
+      const config = {
+        playerVars: {
+          width: 1920,
```

Comment:
hard coded values?

Reply:
removed, this seems to be dynamically evaluated anyway